src/seed.ts

```diff
@@ -1,4 +1,312 @@
+import { removeAdjacentChars } from "skynet-mysky-utils";
+import { hash } from "tweetnacl";
+
+import { dictionary } from "../src/dictionary";
+
 export const SEED_LENGTH = 16;
 export const SEED_WORDS_LENGTH = 13;
 export const CHECKSUM_WORDS_LENGTH = 2;
 export const PHRASE_LENGTH = SEED_WORDS_LENGTH + CHECKSUM_WORDS_LENGTH;
+
+/**
+ * Generates a 15-word seed phrase for 16 bytes of entropy plus 20 bits of checksum. The dictionary length is 1024 which gives 10 bits of entropy per word.
+ *
+ * @returns - The generated phrase.
+ */
+export function generatePhrase(): string {
+  const seedWords = new Uint16Array(SEED_WORDS_LENGTH);
+  window.crypto.getRandomValues(seedWords);
+
+  // Populate the seed words from the random values.
+  for (let i = 0; i < SEED_WORDS_LENGTH; i++) {
+    let numBits = 10;
+    // For the 13th word, only the first 256 words are considered valid.
+    if (i === 12) {
+      numBits = 8;
+    }
+    seedWords[i] = seedWords[i] % (1 << numBits);
+  }
+
+  // Generate checksum from hash of the seed.
+  const checksumWords = generateChecksumWordsFromSeedWords(seedWords);
+
+  const phraseWords: string[] = new Array(PHRASE_LENGTH);
+  for (let i = 0; i < SEED_WORDS_LENGTH; i++) {
+    phraseWords[i] = dictionary[seedWords[i]];
+  }
+  for (let i = 0; i < CHECKSUM_WORDS_LENGTH; i++) {
+    phraseWords[i + SEED_WORDS_LENGTH] = dictionary[checksumWords[i]];
+  }
+
+  return phraseWords.join(" ");
+}
+
+/**
+ * Converts the input seed phrase to the actual seed bytes.
+ *
+ * @param phrase - The input seed phrase.
+ * @returns - The seed bytes.
+ */
+export function phraseToSeed(phrase: string): Uint8Array {
+  phrase = sanitizePhrase(phrase);
+  const [valid, error, seed] = validatePhrase(phrase);
+  if (!valid || !seed) {
+    throw new Error(error);
+  }
+
+  return seed;
+}
+
+/**
+ * Converts the seed bytes to the original phrase. Useful for recovery of a lost phrase.
+ *
+ * @param seed - The seed bytes.
+ * @returns - The original phrase.
+ */
+export function seedToPhrase(seed: Uint8Array): string {
+  const seedWords = seedToSeedWords(seed);
+  return seedWordsToPhrase(seedWords);
+}
+
+/**
+ * Validate the phrase by checking that for every word, there is a dictionary
+ * word that starts with the first 3 letters of the word. For the last word of
+ * the seed phrase (the 12th word), only the first 256 words of the dictionary
```

Comment:
This docstring is confusing.
- Replace `12th` with `13th`. It's the word at index 12 but that's the 13th word.
- Explain why the 13th word is the last word of the seed (e.g. the last two are checksum words).
- Move the 12 to a named constant, e.g. `LAST_WORD_IDX`.